Tests/AblyChatTests/Mocks/MockRealtimeChannel.swift

```diff
@@ -71,7 +74,7 @@ final class MockRealtimeChannel: NSObject, RealtimeChannelProtocol {
     }
 
     var state: ARTRealtimeChannelState {
-        .attached
+        attachResult == .success ? .attached : .failed
```

Comment:
This is too much logic to have in a mock, I think. Let's make this configurable by the tests.

Reply:
done